src/main/java/core/basesyntax/strategy/impl/PurchaseTransactionStrategy.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.db.FruitDB;
+import core.basesyntax.strategy.TransactionProcessor;
+
+public class PurchaseTransactionStrategy implements TransactionProcessor {
+    @Override
+    public void process(FruitTransaction transaction) {
+        String fruit = transaction.getFruit();
+        int quantity = transaction.getQuantity();
+        if (FruitDB.FRUIT_DATA_BASE.containsKey(fruit)) {
+            int currentQuantity = FruitDB.FRUIT_DATA_BASE.get(fruit);
+            if (currentQuantity >= quantity) {
+                FruitDB.FRUIT_DATA_BASE.put(fruit, currentQuantity - quantity);
+            } else {
+                System.out.println("Warning: Not enough " + fruit + " in inventory.");
+            }
+        } else {
+            System.out.println("Warning: " + fruit + " not found in inventory.");
+        }
```

Comment:
```suggestion
        int dbQuantity = FruitDB.FRUIT_DATA_BASE.getOrDefault(fruit, 0);
        int newQuantity = dbQuantity - transactionQuantity;
        if (newQuantity < 0) {
            throw new ...
        }
        FruitDB.FRUIT_DATA_BASE.put(fruit, newQuantity);
```
![image](https://github.com/mate-academy/jv-fruit-shop/assets/20326858/6f941838-ea35-4919-935b-ef9a51bf316e)